changelog/add-ability-for-specify-version-range-for-commandline-commands.dd

```diff
@@ -0,0 +1,8 @@
+Improvment of version specification for commandline commands
+
+Before this release dub was can get only strict version in commands like
+`describe`, `generate`, `fetch` and other. By now version can be specified
+same as dependency version:
+
+    dub fetch foo@'>0.2.0'
+    dub describe foo@'>=0.3.0 <1.0.0'
```

Comment:
```suggestion
All commands now accept a version specification

Before this release dub could only get an exact version for some commands
(`describe`, `generate`, `fetch`, etc...). All commands now accept a version specification,
such as can be found in `dub.json` / `dub.sdl`:

    dub fetch 'foo@>0.2.0'
    dub describe foo@'>=0.3.0 <1.0.0'
    
Note that commands such as `describe` will still not fetch from the network.
```

Is this an accurate description ?

Reply:
Only commands are child for `PackageBuildCommand` are affected: `generate`, `build`, `run`, `test`, `lint`, `describe` (`dustmite` have no version spec). No effects for `clean-caches`, `clean`, `add`, `upgrade`, `fetch`, `install`, and other (pr for `list` https://github.com/dlang/dub/pull/2057).